### PR TITLE
[dagster-airbyte] Support representing normalization-created tables as Assets

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Adds two new optional parameters to `build_airbyte_assets`.

- `upstream_assets` optionally allows a user to specify one or more upstream `AssetKey`s that the Airbyte assets should depend on, this can be used to e.g. represent the source data dependency.
- `normalization_tables` is a Mapping which maps a destination table to a set of one or more derived tables created by the [Airbyte basic normalization](https://docs.airbyte.com/understanding-airbyte/basic-normalization/#nesting) process. These normalization-generated tables will be shown as Assets downstream of their parent table, and materializations will be triggered by the parent table materializing.

Screen Shot 2022-09-09 at 4 14 47 PM

## Test Plan

Tested locally; new unit test.

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9649** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9649" target="_blank">Graphite</a> 
    * **PR #9655** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9655" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/9655?utm_source=stack-comment).

File: python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/test_asset_defs.py [hunk 1]
```diff
@@ -91,3 +92,102 @@ def test_assets(schema_prefix):
...
+                columns=[
+                    TableColumn(name="a", type="str"),
+                    TableColumn(name="b", type="int"),
+                ]
+            ),
+        )
+        in materializations[0].metadata_entries
+    )
+
+    # No metadata for normalized materializations, for now
```
Comment (benpankow) [inline python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/test_asset_defs.py:192]: In a subsequent commit, hoping to add schema information to the normalized tables. We have to do a bit of work to produce this schema.

**Review by OwenKephart — APPROVED** [review]
awesome

**Comment by benpankow** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/9655?utm_source=stack-comment).


Job Reference: ykk4kBl6Nl5kRQqvJjms

**Pull request merged**